main-command/src/main/scala/sbt/Command.scala

```diff
@@ -187,6 +187,13 @@ object Command {
 
   def spacedC(name: String, c: Parser[Char]): Parser[String] =
     ((c & opOrIDSpaced(name)) ~ c.+) map { case (f, rem) => (f +: rem).mkString }
+
+  implicit class CommandWithName(cmd: Command) {
```

Comment:
Could you make it a value class, so we don't have to pay any allocation, please? `extends AnyVal`

Reply:
Not a fan of value class. It doesn't help perf and it has bincompat cost.

Reply:
Value classes are excellent for extension methods, like this one.

Reply:
I think value classes are good in this scenario, they're just syntactic sugar for classes + implicit conversions, and we have lots of those in sbt/sbt.

Reply:
What should it be then?

Reply:
Agreed on using VC for extension methods (also agreed it's not great for other applications). If MiMa says it's ok, then binary compat should be fine.

Reply:
By "bincompat cost", I mean there will be limitation on the future extensibility of the class while keeping the binary compatibility.

Take `RichFile(val asFile: File)` for example -https://github.com/sbt/io/blob/v1.0.2/io/src/main/scala/sbt/io/Path.scala

In sbt 1.1, we support POSIX permission operations via NIO, so it would be nicer if we could expand the parameter of the class to `RichFile(val asFile: File, val asPath: Path)` so that's cached. That kind of extension would not work because value class codegens all methods under companion. As a workaround I implemented POSIX operations in https://github.com/sbt/io/pull/76 as universal trait.

Reply:
If we want to have NIO operations around a (cached) Path we should probably create a `RichPath`. We can dedup methods with universal traits, and the JDK already has the asPath/asFile conversions.